### PR TITLE
Adding in Taint analysis and propagation structures and functions

### PR DESCRIPTION
**Pull request by HAM41**

### CODE REVIEW

File: sim/taint-analysis-structures.h [hunk 1]
```diff
@@ -0,0 +1,16 @@
+typedef struct
```
Comment (phillipstanleymarbell) [inline sim/taint-analysis-structures.h]: There's no need for this to be in a separate file. Please put it in sf.h

File: sim/mkhelp [hunk 2]
```diff
@@ -8,7 +8,7 @@
 
 	echo 'Help Helpstrs[] =\'
 	echo '{\'
-	cat lex-hitachi-sh.c | grep '/\*+' | sort | $1 -F'*|_|}|{|+|,|:' '{print "\t{\n\t\t\t" $2 ",\n\t\t\t\"\tDescription:" $8 "\",\n\t\t\t\"\tArguments:\t" $9 "\",\n\t},"}'
+	cat lex-riscv.c | grep '/\*+' | sort | $1 -F'*|_|}|{|+|,|:' '{print "\t{\n\t\t\t" $2 ",\n\t\t\t\"\tDescription:" $8 "\",\n\t\t\t\"\tArguments:\t" $9 "\",\n\t},"}'
 	echo '};'
-	echo 'int	nHelpstrs ='`cat lex-hitachi-sh.c | grep '/\*+' | wc -l`';'
-	echo 'int	uniqHelpstrs ='`cat lex-hitachi-sh.c | grep '/\*+' | sort -t '"' -k 2 | $1 '{print $2}' | sort | uniq | wc -l`';'
+	echo 'int	nHelpstrs ='`cat lex-riscv.c | grep '/\*+' | wc -l`';'
```
Comment (phillipstanleymarbell) [inline sim/mkhelp]: This change is not part of the taint propagation and should not be in the pull request...

File: sim/mkcommandlist [hunk 3]
```diff
@@ -8,5 +8,5 @@
 
 
 	echo '\section{Simulator Command Set}'
-	cat lex-hitachi-sh.c | grep '/\*+' | sort -t ',' | sed 's/none//' | $1 -F'*|_|}|{|+|,|:|"' '{print tolower($3)","}'
+	cat lex-riscv.c | grep '/\*+' | sort -t ',' | sed 's/none//' | $1 -F'*|_|}|{|+|,|:|"' '{print tolower($3)","}'
```
Comment (phillipstanleymarbell) [inline sim/mkcommandlist]: This change is not part of the taint propagation and should not be in the pull request...

File: sim/machine-hitachi-sh.c [hunk 4]
```diff
@@ -726,6 +726,12 @@ superHresetcpu(Engine *E, State *S)
 	S->MEMSIZE = DEFLT_MEMSIZE;
 	S->MEMBASE = SUPERH_MEMBASE;
 	S->MEMEND = S->MEMBASE + S->MEMSIZE;
+
+	//Initialisation of shadow memory:
```
Comment (phillipstanleymarbell) [inline sim/machine-hitachi-sh.c]: Please follow the coding convention document guidelines for comments.

File: sim/machine-hitachi-sh.c [hunk 5]
```diff
@@ -890,6 +896,15 @@ superHnewstate(Engine *E, double xloc, double yloc, double zloc, char *trajfilen
 		mexit(E, "Failed to allocate memory for S->MEM.", -1);
 	}
 
+	//Initialise array of ShadowMem objects:
```
Comment (phillipstanleymarbell) [inline sim/machine-hitachi-sh.c]: Please follow the coding convention document guidelines for comments.

File: sim/machine-riscv.h [hunk 6]
```diff
@@ -34,14 +34,19 @@
...
 struct RiscvState
 {
 	uint32_t R[RISCV_XMAX];
 	uint64_t fR[RF32FD_fMAX];
 	uint32_t fCSR;
 	RiscvPipe P;
-        uint32_t instruction_distribution[RISCV_OP_MAX]
+        uint32_t instruction_distribution[RISCV_OP_MAX];
+
+	//Taint analysis for registers:
```
Comment (phillipstanleymarbell) [inline sim/machine-riscv.h]: Please follow the coding convention document guidelines for comments.

File: sim/config.h [hunk 7]
```diff
@@ -56,3 +56,4 @@
 #define	SF_FT_TANDEM		0
 #define	SF_BPTS			1
 #define	SF_TRAJECTORIES		1
+#define SF_TAINTANALYSIS	1
```
Comment (phillipstanleymarbell) [inline sim/config.h]: Please follow the coding conventions guidelines for tabbing.

File: sim/Makefile [hunk 8]
```diff
@@ -145,10 +147,10 @@ opstr-riscv.h: decode-ti-msp430.h Makefile
 	./mkopstr-riscv $(GAWK) > opstr-riscv.h
 
 #	Either of the lex-* would suffice for generating the LaTeX / help command array.
-help.h: lex-hitachi-sh.c Makefile
+help.h: lex-riscv.c Makefile
```
Comment (phillipstanleymarbell) [inline sim/Makefile]: This change is not part of the taint propagation and should not be in the pull request...

File: sim/main.c [hunk 9]
```diff
@@ -2043,6 +2046,32 @@ m_sizemem(Engine *E, State *S, int size)
 			"Set memory size to %d Kilobytes\n", S->MEMSIZE/1024);
 	}
 
+	/*Shadow/taintmemory reallocation:
```
Comment (phillipstanleymarbell) [inline sim/main.c]: Please follow the coding convention document guidelines for comments.

File: sim/main.h [hunk 10]
```diff
@@ -510,6 +510,15 @@ struct State
 	int		MEMBASE;
 	int		MEMEND;
 
+	/*Shadow Memory structure allocation
```
Comment (phillipstanleymarbell) [inline sim/main.h]: Please follow the coding convention document guidelines for comments.

File: sim/config.h (hunk 7 above)
Comment (HAM41) [reply sim/config.h]: In my local copy of config.h these all align properly, I'm not sure how to tab so that they align properly online

**Pull request closed without merging**